### PR TITLE
Don't lookup big foreign assets.

### PR DESCRIPTION
**Pull request by winder**
## Summary

We missed the common code path in the previous hotfix, this is the same fix as before but we avoid adding these assets and apps to begin with.

### CODE REVIEW

**Review by cce — APPROVED** [review]
nice! two birds with one stone.

File: accounting/eval_preload.go [hunk 1]
```diff
@@ -34,9 +36,15 @@ func addToCreatorsRequest(stxnad *transactions.SignedTxnWithAD, assetsReq map[ba
 			appsReq[fields.ApplicationID] = struct{}{}
 		}
 		for _, index := range fields.ForeignApps {
+			if index > basics.AppIndex(math.MaxInt64) {
```
Comment (shiqizng) [inline accounting/eval_preload.go:39]: why do we use math.MaxInt64 instead of math.MaxUint64?

Comment (winder) [reply accounting/eval_preload.go:39]: This is related to the `BIGINT` -> `NUMERIC(20)` issue, algod/sqlite has a similar problem that this works around.

**Review by algorandskiy — APPROVED** [review]

**Comment by winder** [comment]
@algobarb helped me run this on the stuck testnet node, it seems to be working.

**Pull request merged**